src/System.Net.Quic/src/System/Net/Quic/QuicConnection.cs

```diff
@@ -20,12 +20,12 @@ public sealed class QuicConnection : IDisposable
         /// <param name="sslClientAuthenticationOptions">TLS options</param>
         /// <param name="localEndPoint">The local endpoint to connect from.</param>
         public QuicConnection(IPEndPoint remoteEndPoint, SslClientAuthenticationOptions sslClientAuthenticationOptions, IPEndPoint localEndPoint = null)
-            : this(remoteEndPoint, sslClientAuthenticationOptions, localEndPoint, implementationProvider: null)
+            : this(null, remoteEndPoint, sslClientAuthenticationOptions, localEndPoint)
```

Comment:
Passing null here causes a null ref, no?

Reply:
Yes, it should, but (a) the only provider we have currently is the mock provider, which is a little weird to make the default; and (b) it was already passing null, so this PR doesn't change anything.

I'm happy to make the mock provider the default, but it does seem a little weird to me. Thoughts?

Reply:
Yeah, even the mock would be a better default than null reffing.
Maybe simplify switching later by adding QuicImplementationProviders.Default so it can be controlled centrally when you add real implementations?

Reply:
Sounds reasonable.